giftree/src/pages/AddGift.tsx

```diff
@@ -0,0 +1,171 @@
+import React, { useState } from 'react';
+import axios from 'axios';
+import { IonContent, IonHeader, IonButtons, IonMenuButton, IonPage, IonTitle, IonToolbar, IonButton, IonLabel, IonList, IonItem, IonInput, IonRow, IonCol } from '@ionic/react';
+import { connect } from '../data/connect';
+
+
+const BASE_URL = 'https://COP4331-1.herokuapp.com/';
+const ENDPOINT_URL = BASE_URL + 'api/addGift';
+
+
+interface StateProps {
+  username?: string;
+  userId?: string;
+}
+
+const AddGift: React.FC<StateProps> = ({ 
+  username,
+  userId
+ }) => {
+  const [giftName, setGiftName] = useState("");
+  const [giftPrice, setGiftPrice] = useState("");
+  const [giftLocation, setGiftLocation] = useState("");
+  const [giftComment, setGiftComment] = useState("");
+  const [giftAdded, setGiftAdded] = useState(false);
+
+  const addGift = async (e: React.FormEvent) => {
+    e.preventDefault();
+    const giftObject = {
+        userId: userId,
+        giftName: giftName,
+        giftPrice: giftPrice,
+        giftLocation: giftLocation,
+        giftComment: giftComment
+    };
+
+    //console.log(giftObject);
+    
+    axios.post(ENDPOINT_URL, giftObject)
+        .then((res) => {
+            console.log(res.data)
+        }).catch((error) => {
+            console.log(error)
+        });
+    setGiftName('');
+    setGiftPrice('');
+    setGiftLocation('');
+    setGiftComment('');
+    setGiftAdded(true);
+  };
+
+  if (giftAdded)
```

Comment:
Regarding the bug that doesn't take you to the "add gift" page after pressing the "Add another gift button" might be because you are not resetting the `giftAdded` variable so whenever it re-renders this page, `giftAdded` is still true, so then it goes into this if statement again.